db/db_range_del_test.cc

```diff
@@ -807,6 +805,122 @@ TEST_F(DBRangeDelTest, IteratorIgnoresRangeDeletions) {
   db_->ReleaseSnapshot(snapshot);
 }
 
+TEST_F(DBRangeDelTest, IteratorRangeTombstoneOverlapsSstable) {
```

Comment:
@ajkr I replaced the other tests I had previously added with this test that covers more cases and allows the easy addition of new test cases. This PR is ready for another look (I'll push a change for the merge conflict shortly).